### PR TITLE
[connectors] Refactor the Zendesk connector

### LINKED ISSUE #8714: [Zendesk] Refactor the connector
Scope:
- Update the signatures of the functions in `zendesk_resources.ts` and `id_conversions.ts` to remove object as parameter when unnecessary.
- Refactor the file structure.

The goal is to spend < 2 hours on this (ideally around 30 min), to not add unnecessary DRYness and to get a super easy-to-review PR.

### PR DESCRIPTION
**Pull request by aubin-tchoi**
## Description

- Close [#8714](https://github.com/dust-tt/dust/issues/8714)
- Goal: refactor in surface the connector, only by moving files around and simplifying signatures. The goal is to do only extremely simple changes like splitting functions between 2 files and changing function signatures from single object to multiple parameters or the inverse.

## Risk

Low.

## Deploy Plan

- Deploy connectors.

### CODE REVIEW

**Comment by aubin-tchoi** [comment]
PMRR

File: connectors/src/connectors/zendesk/index.ts [hunk 1]
```diff
@@ -208,9 +208,7 @@ export class ZendeskConnectorManager extends BaseConnectorManager<null> {
       return new Err(new Error("Connector not found"));
     }
 
-    const brandIds = await ZendeskBrandResource.fetchAllBrandIds({
-      connectorId,
-    });
+    const brandIds = await ZendeskBrandResource.fetchAllBrandIds(connectorId);
```
Comment (flvndvd) [inline connectors/src/connectors/zendesk/index.ts:211]: Why not passing connector directly? To remove ambiguity.

**Review by flvndvd — COMMENTED** [review]
LGTM

File: connectors/src/connectors/zendesk/index.ts (hunk 1 above)
Comment (aubin-tchoi) [reply connectors/src/connectors/zendesk/index.ts:211]: in the methods of the resources here the connectorId sometimes come from a `ConnectorResource`, sometimes from the `ConnectorManager` and are sometimes passed from a Temporal workflow to an activity, I opted for passing only IDs everywhere for consistency

**Pull request merged**